addEntry.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+require_once 'src/database/Database.php';
+
+require_once 'src/entries/EntriesModel.php';
+require_once 'src/languages/LanguagesModel.php';
+require_once 'src/snippets/SnippetsModel.php';
+
+// Form submission processing
+if(isset($_POST['title'])) {
+    $title = $_POST['title'];
+    $description = $_POST['description'];
+
+    $db = connectToDb();
+
+    $entriesModel = new EntriesModel($db);
+    $entrySuccess = $entriesModel->addNewEntry($db, $title, $description);
+
+    // Parse the $POST data for snippet(s), insert to the snippets table if found
+    if($entrySuccess) {
+        $newEntryID = $db->lastInsertId();
+        $snippetIDArray = [];
+
+        // Populate the snippet ID array with unique snippet IDs
+        foreach ($_POST as $key => $value) {
```

Comment:
Ahh really nice way of looping through the snippets

Reply:
This type of logic is exactly the sort of thing that we would want to write tests for.

What you could do is move this functionality (finding the IDs of the snippets) into a function/object that gets passed $\_POST as a param, and then returns the array of snippets.

You could then test it nicely by giving it a fake assoc array that replicates the post data

Reply:
Cheers, makes sense - I've abstracted the function into a new file (entries.php), in case there are more entry processing related functions that could go in it, and have set up unit tests for it.